### PR TITLE
[BAC-126] Lint & prettier

### PR DESCRIPTION
**Pull request by ken-fun**
Does the following:
* Add eslint configs
* Add prettier configs
* Add npm scripts for eslint and prettier
* Run prettier as part of eslint even though this is not recommended by Prettier 🤷 This way you can configure your IDE formatter to use eslint and you'll get both eslint fixes and prettier formatting.

\+ run `npm run fix` and fix 2 eslint issues manually

***

Also this trick was very important for my Eslint plugin in VSCode:

```
ESLint: Restart ESLint Server
```

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  bracketSpacing: true,
+  printWidth: 100,
```
Comment (Chazzzzzzz) [inline .prettierrc.js:3]: 100 seems to be small, 140/150?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+module.exports = {
+  env: {
+    browser: false,
+    es6: true,
+    jest: true,
+    node: true,
+  },
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
```
Comment (Chazzzzzzz) [inline .eslintrc.js:9]: worthwhile too add some command to package.json to do lint

also, can add .husky to do linting

File: .prettierrc.js (hunk 1 above)
Comment (ken-fun) [reply .prettierrc.js:3]: Prettier doesn't work well above 80, they recommend this pretty strongly in their docs.. I'm deciding between 80 and 100.. I don't think anything above 100 is good for readability

File: .eslintrc.js (hunk 2 above)
Comment (ken-fun) [reply .eslintrc.js:9]: See package.json, should already be there. I'll be reorganizing package.json in a separate PR.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "eslint-config-prettier",
+  ],
+  ignorePatterns: ["coverage/**/*", "dist/**/*", "typechain-types/**/*"],
+  rules: {
+    "@typescript-eslint/no-unused-vars": "off", // TODO: Fix.
+    "@typescript-eslint/no-explicit-any": "off", // TODO: Fix.
+    "@typescript-eslint/no-var-requires": "off", // TODO: Fix.
+
+    // TODO: Fix!!!
+    // "@typescript-eslint/no-floating-promises": "error",
```
Comment (AlbertSu123) [inline .eslintrc.js:33]: Are we adding this to eslint or not?

**Review by AlbertSu123 — APPROVED** [review]

File: .eslintrc.js (hunk 3 above)
Comment (ken-fun) [reply .eslintrc.js:33]: We are but not in this PR

**Review by Chazzzzzzz — APPROVED** [review]

**Pull request merged**